python/ray/data/_internal/logical/operators/write_operator.py

```diff
@@ -0,0 +1,35 @@
+from typing import Any, Dict, Optional
+
+from ray.data._internal.logical.interfaces import LogicalOperator
+from ray.data._internal.logical.operators.map_operator import AbstractMap
+from ray.data.datasource.datasource import Datasource
+
+
+class Write(AbstractMap):
+    """Logical operator for write."""
+
+    def __init__(
+        self,
+        input_op: LogicalOperator,
+        datasource: Datasource,
+        ray_remote_args: Optional[Dict[str, Any]] = None,
+        **write_args,
+    ):
+        super().__init__(
+            "Write",
+            input_op,
+            ray_remote_args,
+        )
+        self._datasource = datasource
+        self._write_args = write_args
+        # Always use task to write.
+        self._compute = "tasks"
+        # Take the input blocks unchanged while writing.
+        self._target_block_size = float("inf")
+        # No-op dummy UDF params so for the purpose of operator fusion, the Write
+        # will be treated as Map.
+        self._fn = None
+        self._fn_args = None
+        self._fn_kwargs = None
+        self._fn_constructor_args = None
+        self._fn_constructor_kwargs = None
```

Comment:
These are all specific to `AbstractUDFMap`, not `AbstractMap`, and shouldn't be needed for the `Write` operator, right?
```suggestion
```

Reply:
Yeah, the only reason they are here is to make the *current* operator fusion happy, which I assume you have pending changes there? If not, I can clean this up and change operator fusion part accordingly.

Reply:
Also the "compute" looks good to have so we don't fuse write into actor UDF map, but can fuse read->write. MapOperator has compute param as well.

Reply:
Ah got it, thank you for saving me from merge conflicts! I'd be happy to update this in my PR if you'd like to leave this as is.